### PR TITLE
Fixed dialogmanager bugs on mobile

### PR DESCRIPTION
**Pull request by Steven-David215**
with James' assistance, this time in code, not script execution order

### CODE REVIEW

**Review by markfilliph — APPROVED** [review]

**Review by ValeriaRincon — APPROVED** [review]
Looks great!

File: Assets/Scripts/DialogueSystem/DialogueManager.cs [hunk 1]
```diff
@@ -99,15 +101,39 @@ public void StartConversation(Conversation convo, ConvoFinishedCallback callb)
...
+    {
+        convoIsDelay = true;
+        yield return new WaitForSeconds(0.5f);
+        convoIsDelay = false;
+    }
     public void ReadNext()
     {
+        if (currentConvo == null || convoIsDelay == true)
+        {
+            Debug.Log("Yup, different stuff happened");
```
Comment (DoopweeSLC) [inline Assets/Scripts/DialogueSystem/DialogueManager.cs:123]: should have removed these before submitting

File: Assets/Scripts/DialogueSystem/DialogueManager.cs [hunk 2]
```diff
@@ -99,15 +101,39 @@ public void StartConversation(Conversation convo, ConvoFinishedCallback callb)
         callback = callb;
 
         AudioManager.instance.Play("Hub_DialoguePop");
+
+        if (convoInProgress == false)
+        {
+            Debug.Log("Yup, stuff happened");
```
Comment (DoopweeSLC) [inline Assets/Scripts/DialogueSystem/DialogueManager.cs:107]: same here

File: Assets/Scripts/DialogueSystem/DialogueManager.cs [hunk 3]
```diff
@@ -191,6 +220,16 @@ public void ReadNext()
             button.GetComponent<Image>().sprite = closeDialogueSprite;
         }
     }
+
+    public void ReadNextEditor()
+    {
+        if (convoInProgress == true)
+        {
+            Debug.Log("Yup, stuff happened in editor");
```
Comment (DoopweeSLC) [inline Assets/Scripts/DialogueSystem/DialogueManager.cs:228]: and here

**Pull request merged**